salesagility/SuiteCRM
# Fixed #7304 - ListView: Fix selection count for the "Select All" case

## Description

The check count isn't valid in the "Select All" case and should be ignored.
On the other hand selectCount is only valid in the "Select All" case and
only exact in case the count query is not disabled in the config.

Fix get_num_selected() to work in all cases, add get_num_selected_string()
which appends a "+" in case count queries are disabled which is used in the
UI to signal the user that there could be more entries selected.

To avoid the UI getting out of sync update all selection count related controls in the
update loop and not just the count label (see https://github.com/salesagility/SuiteCRM/issues/4297)

Fixes #7304

(Also fixes #4297 which I can still reproduce)

## Motivation and Context

The UI should reflect the actual number of selected records.

## How To Test This

* Go to the leads module
* Select all records
* The "Selected: X" label should be correct
* Bulk Action > Delete -> The message should show the correct count

## Types of changes

### Final checklist
